### DISCUSSION TITLE
Can we have a native receiver for NATS message queue similar to KAFKA and RabbitMQ.

### DISCUSSION THREAD

**Discussion by vikramrajsivaram** [open-telemetry/opentelemetry-collector-contrib#24266] [Ideas]
Do we have a plan to bring out a native NATS receiver similar to KAFKA and RabbitMQ. Currently we have a NATS surveyor which directly exports data to Prometheus

**Comment by c16a** [comment]
This seems like a good idea and I'm interested to contribute.

**Comment by vikramrajsivaram** [comment]
Found this in Grafana forum https://grafana.com/docs/grafana-cloud/send-data/metrics/metrics-prometheus/prometheus-config-examples/nats-authors-nats/

**Comment by hwinkel** [comment]
What about to ship the telemetry data over NATS. Creating a p2p topology for many collectors in a medium complex topology is more complex than have NATS topology established once and use it for all sort of things like telemetry shipping

**Reply by dennisme** [reply]
It can be done with kafka now, it works great. +1 to nats.
